### DISCUSSION TITLE
New user signup with email and OTP without password

### DISCUSSION THREAD

**Discussion by cananda** [supabase/supabase#9433] [Questions] [answered]
Hello!  I would like users to not use password and use email with OTP. As per docs, there is no signup with OTP and the signup requires a password to be set - https://supabase.com/docs/reference/javascript/next/auth-signup

How would I go about not having users rely on passwords to signup for my application

**Comment by GaryAustin1** [comment, marked as answer]
It is not documented  but on V1 you use signIn for OTP or magic link. The first time an email is used a user is created.  To use OTP you put {{Token}} in the email template.  Then call this with the Token:

const { session, error: verifyError } = await supabase.auth.verifyOTP({
  email,
  token,
  type: 'magiclink'
})

**Reply by cananda** [reply]
@GaryAustin1 I'm on v2 rc SDK and not keen on going back to v1.  I tried this - 

```
		const { data, error } = await supabase.auth.signInWithOtp({
			email: email,
            options: {
                shouldCreateUser: true
            }
		});
```
User is created.  Then I try login with the same code above, I receive the OTP via email. And I verify OTP with,

```
		const { data, error } = await supabase.auth.verifyOtp({
			email: email,
			token: token,
			type: 'magiclink'
		});
```

However, I receive as response - msg | "Token has expired or is invalid"

Any pointers on where I'm going wrong?

**Reply by GaryAustin1** [reply]
That is odd....  This is my signin and I set the type to use in the verifyOtp call in $confirm
```
 async function resetPass() {
        const { data, error } = await supabase.auth.signInWithOtp({
            email:email
            }
        )
        $confirm = 'magiclink'
   }
```
```
if (confirmValue) {
            const {data, error} = await supabase.auth.verifyOtp({
                email: email,
                token: confirmValue.trim(),
                type: $confirm
            })
```

So your code looks right at quick glance.   The otp expires by default in one hour I think, setting in Authentication/email section.

**Comment by cananda** [comment]
This is working now, not sure what changed, mostly it's my environment I guess :) Thank you for your help
